app/src/main/java/com/cornellappdev/android/eateryblue/data/models/Eatery.kt

```diff
@@ -188,6 +189,103 @@ data class Eatery(
+
+            val timeString = "$openTime - $closeTime"
+
+            if (dayOfWeek != null && dailyHours[dayOfWeek]?.none { it.contains(timeString) } != false) {
+                dailyHours.computeIfAbsent(dayOfWeek) { mutableListOf() }.add(timeString)
+            }
+        }
+
+        DayOfWeek.values().forEach { dayOfWeek ->
+            dailyHours.computeIfAbsent(dayOfWeek) { mutableListOf("Closed") }
+        }
+
+        val groupedHours = dailyHours.entries.groupBy({ it.value }, { it.key })
+
+        return groupedHoursFormatHelper(groupedHours)
+    }
+
+    private val dayOrder = mapOf(
+        "Sunday" to 1,
+        "Monday" to 2,
+        "Tuesday" to 3,
+        "Wednesday" to 4,
+        "Thursday" to 5,
+        "Friday" to 6,
+        "Saturday" to 7
+
+    )
+
+    private fun groupedHoursFormatHelper(groupedHours: Map<MutableList<String>, List<DayOfWeek>>): List<Pair<String, List<String>>> {
+        val formattedHours = LinkedHashMap<String, List<String>>()
+
+        groupedHours.forEach { entry ->
+            val days = entry.value.sortedBy {
+                val dayName = "$it".take(1).uppercase() + "$it".drop(1).lowercase()
+                dayOrder[dayName] ?: Int.MAX_VALUE
+            }
+            var curStrings = mutableListOf<String>()
+            for (i in (days.indices)) {
+                val curDay = "${days[i]}".take(1).uppercase() + "${days[i]}".drop(1).lowercase()
+                if (i == days.size - 1) {
+                    curStrings.add(curDay)
+                    val formattedString = formatString(curStrings)
+                    formattedHours[formattedString] = entry.key
+                } else {
+                    curStrings.add(curDay)
+                    if (!isNext("${days[i]}", "${days[i + 1]}")) {
+                        val formattedString = formatString(curStrings)
+                        formattedHours[formattedString] = entry.key
+                        curStrings = mutableListOf()
+                    }
+                }
+            }
+        }
+
+        var formattedHoursList = formattedHours.toList().sortedBy { entry ->
+            val firstDay = entry.first.split(" to ", " ", limit = 2).first()
+            dayOrder[firstDay] ?: Int.MAX_VALUE
+        }
+        return formattedHoursList
+    }
+
+    private fun isNext(day1: String, day2: String): Boolean {
+        if (day1 == "MONDAY") {
+            return day2 == "TUESDAY"
+        } else if (day1 == "TUESDAY") {
+            return day2 == "WEDNESDAY"
+        } else if (day1 == "WEDNESDAY") {
+            return day2 == "THURSDAY"
+        } else if (day1 == "THURSDAY") {
+            return day2 == "FRIDAY"
+        } else if (day1 == "FRIDAY") {
+            return day2 == "SATURDAY"
+        } else if (day1 == "SATURDAY") {
+            return false
+        } else if (day1 == "SUNDAY") {
+            return day2 == "MONDAY"
+        } else {
+            return false
+        }
```

Comment:
You can turn this into a `when` block instead of using `if-else`. This makes the code look a bit more readable.

IntelliJ actually can automatically do this for you; just accept this suggestion:

![image](https://github.com/cuappdev/eatery-blue-android/assets/47724806/e40a2596-15e4-4fda-a0d7-0b2a9fb4a1d3)

Reply:
I can also foresee intelliJ immediately afterwards suggesting converting this into a one-line return statement. Make sure to do that too; basically, you want to return the result of the `when` block.